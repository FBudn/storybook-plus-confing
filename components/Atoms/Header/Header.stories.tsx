import React from "react";
import { View } from "react-native";
import Header from "./Header";

const HeaderMeta = {
  title: "Atoms/Header",
  component: Header,
  args: {
    testId: "test-header-id",
    children: "Header jestem 2",
  },
  decorators: [
    (Story: any) => (
      <View
        style={{
          alignItems: "center",
          justifyContent: "center",
          flex: 1,
          backgroundColor: "white",
        }}
      >
        <Story />
      </View>
    ),
  ],
};

export default HeaderMeta;

export const Basic = {};
