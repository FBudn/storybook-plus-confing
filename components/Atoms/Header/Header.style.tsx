import { StyleSheet } from "react-native";

const HeaderStyles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
  },
  text: {
    fontWeight: "bold",
    color: "rgb(82 82 91)",
  },
});

export default HeaderStyles;
