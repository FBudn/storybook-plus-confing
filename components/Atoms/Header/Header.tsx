import React from "react";
import { StyleSheet, View, Text } from "react-native";

export interface HeaderProps {
  children: string;
  testId?: string;
}

const Header: React.FC<HeaderProps> = ({ children, testId }) => {
  return (
    <View style={styles.container} testID={testId}>
      <Text style={styles.text}> {children} </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "yellow",
  },
  text: {
    fontWeight: "bold",
    color: "rgb(82 82 91)",
  },
});

export default Header;
