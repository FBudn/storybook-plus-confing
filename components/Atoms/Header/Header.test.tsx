import React from "react";
import { render, screen } from "@testing-library/react-native";
import TestRenderer from "react-test-renderer";
import Header, { HeaderProps } from "./Header";

const props: HeaderProps = {
  children: `Header Test`,
  testId: `test-header-id`,
};

test("Should render Header component", () => {
  render(<Header {...props} />);
  const HeaderElement = screen.getByTestId(`test-header-id`);
  expect(HeaderElement).toBeOnTheScreen();
});

test("Should match snapshot:", () => {
  const HeaderElement = TestRenderer.create(<Header {...props} />).toJSON();
  expect(HeaderElement).toMatchSnapshot();
});

test("Should render and have styles:", () => {
  render(<Header {...props} />);
  const HeaderElement = screen.getByTestId(`test-header-id`);
  expect(HeaderElement).toBeOnTheScreen();

  expect(HeaderElement).toHaveStyle({
    flex: 0.05,
    flexDirection: "row",
    alignItems: "center",
  });
});

test(`Should render with text`, () => {
  render(<Header {...props} />);
  const HeaderElement = screen.getByTestId(`test-header-id`);
  expect(HeaderElement).toBeOnTheScreen();
  expect(HeaderElement).toHaveTextContent(`Header Test`);
});
