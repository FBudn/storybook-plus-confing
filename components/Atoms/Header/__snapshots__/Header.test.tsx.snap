// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot: 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
  testID="test-header-id"
>
  <Text
    style={
      {
        "color": "rgb(82 82 91)",
        "fontWeight": "bold",
      }
    }
  >
     
    Header Test
     
  </Text>
</View>
`;
